src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -2190,6 +2103,227 @@ private FeedbackSessionResultsBundle getFeedbackSessionResultsForUserWithParams(
         return results;
     }
 
+
+
+    private FeedbackSessionResultsBundle getResponsesForQuestion(String feedbackSessionName,
+            String courseId, String userEmail, UserType.Role role, CourseRoster roster,
+            Map<String, String> params, FeedbackSessionAttributes session,
+            List<FeedbackQuestionAttributes> allQuestions, List<FeedbackResponseAttributes> responses,
+            Map<String, FeedbackQuestionAttributes> relevantQuestions, Map<String, String> emailNameTable,
+            Map<String, String> emailLastNameTable, Map<String, String> emailTeamNameTable,
+            Map<String, Set<String>> sectionTeamNameTable, Map<String, boolean[]> visibilityTable,
+            Map<String, List<FeedbackResponseCommentAttributes>> responseComments,
+            FeedbackSessionResponseStatus responseStatus, boolean isIncludeResponseStatus, String section,
+            String questionId, boolean isQueryingResponseRateStatus) {
+        
+        if (isQueryingResponseRateStatus) {
+            responseStatus = section == null && isIncludeResponseStatus
+                           ? getFeedbackSessionResponseStatus(session, roster, allQuestions)
+                           : null;
+        } else {
+            InstructorAttributes instructor = null;
+            if (role == Role.INSTRUCTOR) {
+                instructor = instructorsLogic.getInstructorForEmail(courseId, userEmail);
+            }
+            FeedbackQuestionAttributes question = fqLogic.getFeedbackQuestion(questionId);
+            if (question != null) {
+                relevantQuestions.put(question.getId(), question);
+                
+                List<FeedbackResponseAttributes> responsesForThisQn =
+                        getVisibleResponsesForQuestion(userEmail, role, section, session, question);
+   
+                boolean thisQuestionHasResponses = !responsesForThisQn.isEmpty();
+                if (thisQuestionHasResponses) {
+                    for (FeedbackResponseAttributes response : responsesForThisQn) {
+                        boolean isVisibleResponse = false;
+                        if (response.giver.equals(userEmail)
+                                || isResponseVisibleForUser(userEmail, role, null, new HashSet<String>(),
```

Comment:
The original functionality never properly took students into account, but this was masked by the fact that all calls to the main `getFeedbackSessionResultsForUserWithParams()` method passed in `role` as an instructor. It might be possible to restore functionality for students in the next pass (if required).